rust/worker/src/blockstore/memory/reader_writer.rs

```diff
@@ -11,6 +11,16 @@ pub(crate) struct MemoryBlockfileWriter {
     id: uuid::Uuid,
 }
 
+pub(crate) struct MemoryBlockfileFlusher {
+    id: uuid::Uuid,
+}
+
+impl MemoryBlockfileFlusher {
```

Comment:
I'm a little confused, where is flush() for the MemoryBlockfileFlusher implemented?

Reply:
Why is that needed? 

```
impl SegmentFlusher for RecordSegmentFlusher {
    async fn flush(self) -> Result<HashMap<String, Vec<String>>, Box<dyn ChromaError>> {
        let user_id_to_id_bf_id = self.user_id_to_id_flusher.id();
        let id_to_user_id_bf_id = self.id_to_user_id_flusher.id();
        let id_to_data_bf_id = self.id_to_data_flusher.id();
        let max_offset_id_bf_id = self.max_offset_id_flusher.id();
        let res_user_id_to_id = self.user_id_to_id_flusher.flush::<&str, u32>().await;
        let res_id_to_user_id = self.id_to_user_id_flusher.flush::<u32, &str>().await;
        let res_id_to_data = self.id_to_data_flusher.flush::<u32, &DataRecord>().await;
        let res_max_offset_id = self.max_offset_id_flusher.flush::<&str, u32>().await;

        let mut flushed_files = HashMap::new();
        ...
```
The flusher that `commit()` returns contains the id which is extracted as in above. Then individual `flush` for each blockfile is called which performs the actual flush. So, for in-memory blockfile this flush should be a no-op which already exists

Reply:
but then how are any file ids returned?

Reply:
yep got it sorry, got mixed up between blockfile flush and segment flush